File: catalogue-api/src/main/java/io/quartic/catalogue/api/CloudGeoJsonDatasetLocator.java
package io.quartic.catalogue.api;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.quartic.common.SweetStyle;
import org.immutables.value.Value;

@SweetStyle
@Value.Immutable
@JsonSerialize(as = CloudGeoJsonDatasetLocatorImpl.class)
@JsonDeserialize(as = CloudGeoJsonDatasetLocatorImpl.class)
public interface CloudGeoJsonDatasetLocator extends DatasetLocator {
    String path();
}

File: catalogue-api/src/main/kotlin/io/quartic/catalogue/api/CloudGeoJsonDatasetLocator.kt
package io.quartic.catalogue.api

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import io.quartic.common.SweetStyle
import org.immutables.value.Value

data class CloudGeoJsonDatasetLocator(val path: String, val streaming: Boolean = false) : DatasetLocator
